src/main/java/com/example/shorturl/controller/login/AuthController.java

```diff
@@ -0,0 +1,54 @@
+package com.example.shorturl.controller.login;
+
+import com.example.shorturl.model.user.UserEntity;
+import com.example.shorturl.model.user.UserRequest;
+import com.example.shorturl.model.user.UserResponse;
+import com.example.shorturl.service.user.UserService;
+import com.example.shorturl.utils.exceptions.UserException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
+
+import javax.validation.Valid;
+import java.util.List;
+import java.util.Objects;
+
+@RestController
+@RequestMapping("api/auth")
+public class AuthController {
+
+    private final UserService userService;
+
+    public AuthController(final UserService userService){
+        this.userService = userService;
+    }
+
+    @PostMapping("/signup")
+    public ResponseEntity<UserResponse> signUp(@Valid @RequestBody UserRequest userRequest) throws UserException {
+
+        if(Objects.isNull(userRequest))
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "User request is missing or empty");
+
+        List<UserEntity> userListObtained = userService.findUserByUserNameOrEmail(userRequest.getUserName(), userRequest.getEmail());
+
+        if(userListObtained.size() > 0)
+            userListObtained.stream().forEach( user -> {
+                if(user.getUserName().equals(userRequest.getUserName()))
+                    throw new ResponseStatusException(HttpStatus.GONE, "user name already used");
```

Comment:
Why this is GONE, I think, it should be Bad Request
Also, about this method
I have some doubts about this implementation
Why do you return list of users? You have to return one user everytime and not to break stateless state of this endpoint

Reply:
I found that the userName can be equal to one user and the email to another. That is why I returned a list, because in the worst scenario I will have 2 users with the "same" information

Reply:
You should have a unique email and unique username and you shouldn't care about how many users are in it
Just go to the database and check for this email/user within one request and if you find some -> return bad exception with message "Username or email already exists"

Reply:
Ok, sure